### PR TITLE
tune_skin: add adjust_max_skin feature.

### LINKED ISSUE #2961: s.cell_system.tune_skin() ignores short range cutoff
The maximum skin considered must be curtailed such that
r_c +skin <0.5 l
where r_c = short range cutoff, l = local box length.
Alternatively, it needs to throw a clear error msg.
To reproduce:

```
import espressomd
s=espressomd.System(box_l=[1,1,1])
s.time_step=0.01
s.non_bonded_inter[0,0].lennard_jones.set_params(epsilon=1,sigma=0.2,cutoff=0.3,shift="auto")
s.cell_system.tune_skin(min_skin=0.1,max_skin=0.6,tol=0.05, int_steps=3)
```

### PR DESCRIPTION
**Pull request by pkreissl**
Fixes #2961

Description of changes:
 - Add `adjust_max_skin` parameter to `tune_skin()` method, which adds the option to automatically adjust `max_skin` if the passed value is larger than permitted (0.5 * local box length - short range cutoff)
 - Rename core variables to match interface.

As a test you can run the following script adjusted from @RudolfWeeber s minimal script in issue #2961
```python
import espressomd
s = espressomd.System(box_l=[1,1,1])
s.time_step = 0.01
s.non_bonded_inter[0,0].lennard_jones.set_params(epsilon=1,sigma=0.2,cutoff=0.3,shift="auto")
print("=> Tune with adjustment:")
s.cell_system.tune_skin(min_skin=0.1,max_skin=0.6,tol=0.05, int_steps=3, adjust_max_skin=True)
print(s.cell_system.skin)
print("=> and fail without:")
s.cell_system.tune_skin(min_skin=0.1,max_skin=0.6,tol=0.05, int_steps=3)
``` 
resulting in the following output:
```
$ ./pypresso tune_skin.py
=> Tune with adjustment:
0.17500000000000002
=> and fail without:
There were unhandled errors.
ERROR: number of cells 1 is smaller than minimum 8 (interaction range too large or min_num_cells too large)
```

### CODE REVIEW

File: src/python/espressomd/cellsystem.pyx [hunk 1]
```diff
@@ -296,11 +296,15 @@ cdef class CellSystem(object):
...
+                            to the maximum permissible skin (in case the passed
+                            value is too large). Set to ``False`` by default.
 
         Returns
         -------
         :attr:`skin`
 
         """
-        c_tune_skin(min_skin, max_skin, tol, int_steps)
+        c_tune_skin(min_skin, max_skin, tol, int_steps, int(adjust_max_skin))
```
Comment (jngrad) [inline src/python/espressomd/cellsystem.pyx]: Why `int` here and in the signatures in `cellsystem.pxd` and `tuning.pxd` instead of `bool`?

Comment (pkreissl) [reply src/python/espressomd/cellsystem.pyx]: You're right, this is not necessary. Just a leftover from debugging I overlooked...

**Comment by RudolfWeeber** [comment]
the "unhandled error" indicates that at the end of the tune_skin function in python utils.handle_errors() needs to be called. That would then raise an exception

Can you please add a test based on the script in the comment.
You can check for the exception via with self.assertRaises()


Please set adjust_max_skin=True in maintainer/benchmarks/p3m.py. That may solve #2924.

**Comment by pkreissl** [comment]
Now, with proper error handling
```
$ ./pypresso tune_skin.py
=> Tune with adjustment:
0.17500000000000002
=> and fail without:
ERROR: number of cells 1 is smaller than minimum 8 (interaction range too large or min_num_cells too large)
Traceback (most recent call last):
  File "tune_skin.py", line 9, in <module>
    s.cell_system.tune_skin(min_skin=0.1,max_skin=0.6,tol=0.05, int_steps=3)
  File "cellsystem.pyx", line 310, in espressomd.cellsystem.CellSystem.tune_skin
  File "utils.pyx", line 265, in espressomd.utils.handle_errors
Exception: Error during tune_skin: ERROR: number of cells 1 is smaller than minimum 8 (interaction range too large or min_num_cells too large)
```

File: testsuite/python/CMakeLists.txt [hunk 2]
```diff
@@ -77,6 +77,7 @@ foreach(TEST_COMBINATION lb.cpu-p3m.cpu-lj;lb.gpu-p3m.cpu-lj;ek.gpu)
   endforeach(TEST_BINARY)
 endforeach(TEST_COMBINATION)
 python_test(FILE cellsystem.py MAX_NUM_PROC 4)
+python_test(FILE tune_skin.py MAX_NUM_PROC 1)
```
Comment (jngrad) [inline testsuite/python/CMakeLists.txt:80]: @RudolfWeeber how many threads should we use here?

**Review by RudolfWeeber — APPROVED** [review]
bors r+

**Comment by jngrad** [comment]
bors r=RudolfWeeber

**Comment by jngrad** [comment]
bors retry

**Pull request merged**